frontend/src/components/pages/CreateReview/BookModal.tsx

```diff
@@ -75,30 +84,61 @@ const BookModal = (props: BookModalProps): React.ReactElement => {
     return years.reverse();
   };
 
-  /** Clears book data */
-  const clearBookData = () => {
-    setTitle("");
-    setCoverImage("");
-    setPrefix("");
-    setSeriesOrder("");
-    setIllustrator([]);
-    setTranslator([]);
-    setFormat("");
-    setGenre("");
-    setMinAge(0);
-    setMaxAge(0);
-    setAuthors([]);
-    setPublisher("");
-    setSeriesName("");
-    setIsbn("");
-    setPrice(0);
-    setPublicationYear("");
-  };
-
   // Calls clearBookData() whenever modal is closed
   useEffect(() => {
-    if (isOpen === false) clearBookData();
-  }, [isOpen]);
+    /** Clears book data */
+    const clearBookData = () => {
+      setCurrBook(null);
```

Comment:
Do you think it's worth having both a `currBook` and every field of the Book individually in state?